content/src/main/content/jcr_root/apps/acs-commons/touchui-widgets/showhidedialogfields/source/showhidedialogfields.js

```diff
@@ -0,0 +1,103 @@
+ * selection made in the dropdown/select or on checkbox check. 
+ *
+ * How to use:
+ * 1. Select Field
+ * - add the empty property cq-dialog-dropdown-showhide to the dropdown/select element
+ * - add the data attribute cq-dialog-showhide-target to the dropdown/select element, value should be the
+ *   selector, usually a specific class name, to find all possible target elements that can be shown/hidden.
+ * - add the target class to each target component that can be shown/hidden
+ * - add the class hidden to each target component to make them initially hidden
+ * - add the attribute showhidetargetvalue to each target component, the value should equal the value of the select
+ *   option that will unhide this element.
+ *   
+ * 2. Checkbox Field
+ * - add the empty property cq-dialog-checkbox-showhide to the checkbox element
+ * - add the data attribute cq-dialog-showhide-target to the checkbox element, value should be the
+ *   selector, usually a specific class name, to find all possible target elements that can be shown/hidden.
+ * - add the target class to each target component that can be shown/hidden
+ * - add the class hidden to each target component to make them initially hidden
+ * - add the attribute showhidetargetvalue to each target component, the value should equal to:
+ * 	 'true', if the field is to be displayed when Checkbox is selected.
+ * 	 'false', if the field is to be displayed when Checkbox is unselected.
+ */
+
+(function(document, $) {
+    "use strict";
+
+    // when dialog gets injected
+    $(document).on("foundation-contentloaded", function(e) {
+        // if there is already an inital value make sure the according target element becomes visible
+        showHide($("[data-cq-dialog-dropdown-showhide], [data-cq-dialog-checkbox-showhide]", e.target));
+    });
+
+    $(document).on("selected", "[data-cq-dialog-dropdown-showhide]", function(e) {
+        showHide($(this));
+    });
+
+    $(document).on("change", "[data-cq-dialog-checkbox-showhide]", function(e) {
+    	showHide($(this));
+    });
+
+    function showHide(el){
+
+        el.each(function(i, element) {
+
+            var target, value;
+            var type = getFieldType(element);
+            
+            switch (type) {
+		        case "select":
+		        	var widget = $(element).data("select");
+		        	if (widget) {	                
+			        	// get the selected value
+		                value =  widget.getValue();
+		        	}
+	                break;
+		        case "checkbox":
+		        	// get the selected value
+		        	value =  $(element).prop('checked');
+            }
+            
+            // get the selector to find the target elements. its stored as data-.. attribute
+        	target = $(element).data("cq-dialog-showhide-target");
+            
+            if (target) {
+            	hideUnselectedElements(target, value);
```

Comment:
Just double checking that's it's fine if type doesn't match select or checkbox and value never gets set ...  the I don't see how it would happen based on the events but I've seen weird stuff in touch UI impls. Will merge but if it's - possibl problem just submit an update.